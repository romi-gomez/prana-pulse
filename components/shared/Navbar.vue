<template>
  <nav class="bg-main-dark text-white p-4">
    <ul class="flex space-x-4">
      <li><router-link to="/" class="hover:underline">Home</router-link></li>
      <li><router-link to="/dashboard" class="hover:underline">Profile</router-link></li>
      <li><router-link to="/oracles" class="hover:underline">Oracles</router-link></li>
      <li><router-link to="/settings" class="hover:underline">Settings</router-link></li>
      <li><router-link to="/login" class="hover:underline">Login</router-link></li>
      <li><router-link to="/signup" class="hover:underline">Signup</router-link></li>
       
    </ul>
  </nav>
</template>

<script setup>
// No additional script needed for this basic structure
</script>

<style scoped>
</style>