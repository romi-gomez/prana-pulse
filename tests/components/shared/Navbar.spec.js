import { mount } from '@vue/test-utils'
import Navbar from '@/components/shared/Navbar.vue'
import { RouterLinkStub } from '@vue/test-utils'

describe.only('Navbar', () => {
  test('renders correctly', () => {
    const wrapper = mount(Navbar, {
      global: {
        stubs: {
          RouterLink: RouterLinkStub
        }
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('contains navigation links', () => {
    const wrapper = mount(Navbar, {
      global: {
        stubs: {
          RouterLink: RouterLinkStub
        }
      }
    })
    const links = wrapper.findAllComponents(RouterLinkStub)
    expect(links.length).toBe(6)
    expect(links[0].text()).toBe('Home')
    expect(links[1].text()).toBe('Profile')
    expect(links[2].text()).toBe('Oracles')
    expect(links[3].text()).toBe('Settings')
    expect(links[4].text()).toBe('Login')
    expect(links[5].text()).toBe('Signup')
  })
})