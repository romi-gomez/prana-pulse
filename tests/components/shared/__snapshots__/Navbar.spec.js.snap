// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  class="bg-main-dark text-white p-4"
>
  <ul
    class="flex space-x-4"
  >
    <li>
      <a
        class="hover:underline"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class="hover:underline"
      >
        Profile
      </a>
    </li>
    <li>
      <a
        class="hover:underline"
      >
        Oracles
      </a>
    </li>
    <li>
      <a
        class="hover:underline"
      >
        Settings
      </a>
    </li>
    <li>
      <a
        class="hover:underline"
      >
        Login
      </a>
    </li>
    <li>
      <a
        class="hover:underline"
      >
        Signup
      </a>
    </li>
  </ul>
</nav>
`;
